src/Lean/Compiler/IR/EmitLLVM.lean

```diff
@@ -1479,6 +1488,13 @@ def emitMainFn (mod : LLVM.Module llvmctx) (builder : LLVM.Builder llvmctx) : M
   let inslot ← buildPrologueAlloca builder (← LLVM.pointerType inty) "in"
   let resty ← LLVM.voidPtrType llvmctx
   let res ← buildPrologueAlloca builder (← LLVM.pointerType resty) "res"
+
+  let argcval ← LLVM.getParam main 0
+  let argvval ← LLVM.getParam main 1
+  let argcval ← LLVM.buildSextOrTrunc builder argcval (← LLVM.i32Type llvmctx)
+  let value ← callLeanSetupLibUV builder argcval argvval
+  LLVM.buildStore builder value argvval
```

Comment:
Should this `value` return value be used in place of `argvval` below?

Reply:
It looks to me like `uv_setup_args` is permitted to mangle its input, and the caller is expected to use only the output after calling it.

Reply:
Fixed.